app/auth/components/Navbar.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Switch, Route, Link } from 'react-router-dom';
+import { Menu } from 'semantic-ui-react';
+
+const Navbar = () => (
+  <Menu attached size="huge">
+    <Menu.Menu>
+      <Menu.Item header as={Link} to="/">
+        Internship Portal
+      </Menu.Item>
+    </Menu.Menu>
+    <Menu.Menu position="right">
+      <Switch>
+        <Route exact path="/login">
+          <Menu.Item as={Link} to="/signup">
+            Sign Up
+          </Menu.Item>
+        </Route>
+        <Route exact path="/signup">
+          <Menu.Item as={Link} to="/login">
+            Login
+          </Menu.Item>
+        </Route>
```

Comment:
I don't think you need to match with the `Router` for rendering these links dynamically. This should be taken care by a `prop`.

Reply:
What do you mean by 'matching with the Router'?

Reply:
I guess he is showing `login` link in `sign up` page, and he is achieving that by matching with the url from router. See `<Route exact path="/signup">. I just think this can easily be achieved by a `prop` sent from the parent.